Unit tests fixed

hrobeers opened the pull request:
I fixed the unit tests.
They now compile and run successfully for the ppcoin code.
I'll leave a note on peer4commit just in case you like this contribution.

belovachap commented on src/test/transaction_tests.cpp:
```diff
@@ -144,4 +142,29 @@ BOOST_AUTO_TEST_CASE(test_GetThrow)
+    t.vin.resize(3);
+    t.vin[0].prevout.hash = dummyTransactions[0].GetHash();
+    t.vin[0].prevout.n = 0;
+    t.vin[1].prevout.hash = dummyTransactions[1].GetHash();;
```
extra trailing semicolons mean anything?

hrobeers commented:
Nope, they have no meaning. They don't hurt either.
They were alreasy in the code.
But it's indeed a good time to clean this up, so I fixed it and updated the PR.

Thanks

belovachap commented:
lgtm :+1: pretty important fix!

sunnyking commented:
hi hrobeers,

The changes with net.cpp and util.cpp needs more validation, are these changes from recent bitcoin code? Do you require those with the test fixes?

Also, it looks like in the transaction test an object was renamed from t1 to t? Generally speaking beautifying bitcoin code should be avoided as it would cause unnecessary maintenance work when attempting to merge with bitcoin code.

hrobeers commented:
Hi sunny,

The net.cpp change can is coming straight upstream bitcoin (https://github.com/bitcoin/bitcoin/blob/0.12/src/net.cpp#L1307) and is needed to compile with the latest version of miniupnpc (which come standard on ArchLinux). So you can safely ignore this change if you don't want to support these versions.

About util.cpp, this change is also coming from upstream bitcoin (https://github.com/bitcoin/bitcoin/blob/0.8/src/util.cpp#L247) and is needed to avoid log messages from obfuscating the test output.

About the t1 naming, as you can see in the bitcoin code. Sometimes t1 is used and sometimes t is used. I decided to name it consistently to t, but I can get it all aligned to the bitcoin code if you prefer so.

Please let me know what you think and I'll update the pull request.

Thanks,
hrobeers

sunnyking commented:
Thanks! Regarding the util.cpp (windows print to debugger) and net.cpp (latest upnp) changes, could you make them into two separate backport commits respectively, commenting on the source of each backport with original bitcoin commit number, then they are good to get into main.
Please also revert the renaming in test code and then group the test changes in single or multiple commits as you prefer.
Thanks a lot for the good work :)
